From 38bb137e9923a4c4c64befcf8b5fda29691440e0 Mon Sep 17 00:00:00 2001
From: TADANO Tokumei <aimingoff@pc.nifty.jp>
Date: Thu, 23 Feb 2023 19:17:17 +0900
Subject: [PATCH] lavf/mpegts.c: set some properties for ARIB caption

Some additional properties are set for ARIB caption.
* need_parsing = 0
  ARIB caption doesn't require any parser.
  This avoids "parser not found" warning message.
* need_context_update = 1
  When any profiles are changed, set this flag to notify.

Signed-off-by: rcombs <rcombs@rcombs.me>
---
 libavformat/mpegts.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/libavformat/mpegts.c b/libavformat/mpegts.c
index a8fa3c0b28cb..385d78b98bb3 100644
--- a/libavformat/mpegts.c
+++ b/libavformat/mpegts.c
@@ -2176,8 +2176,12 @@ int ff_parse_mpeg2_descriptor(AVFormatContext *fc, AVStream *st, int stream_type
 
             st->codecpar->codec_type = AVMEDIA_TYPE_SUBTITLE;
             st->codecpar->codec_id   = AV_CODEC_ID_ARIB_CAPTION;
-            st->codecpar->profile    = picked_profile;
+            if (st->codecpar->profile != picked_profile) {
+                st->codecpar->profile = picked_profile;
+                sti->need_context_update = 1;
+            }
             sti->request_probe = 0;
+            sti->need_parsing = 0;
         }
         break;
     case 0xb0: /* DOVI video stream descriptor */
